src/api/ha_civo/controlplane.go

```diff
@@ -0,0 +1,109 @@
+package ha_civo
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/civo/civogo"
+)
+
+// scriptWithoutCP_1 script used to configure the control-plane-1 with no need of output inital
+// params
+//
+//	dbEndpoint: database-Endpoint
+//	privateIPlb: private IP of loadbalancer
+func scriptWithoutCP_1(dbEndpoint, privateIPlb string) string {
+	return fmt.Sprintf(`#!/bin/bash
+export K3S_DATASTORE_ENDPOINT='%s'
+curl -sfL https://get.k3s.io | sh -s - server --node-taint CriticalAddonsOnly=true:NoExecute --tls-san %s
+`, dbEndpoint, privateIPlb)
+}
+
+func scriptWithCP_1() string {
+	return `#!/bin/bash
+cat /var/lib/rancher/k3s/server/token
+`
+}
+
+func scriptCP_n(dbEndpoint, privateIPlb, token string) string {
+	return fmt.Sprintf(`#!/bin/bash
+export SECRET='%s'
+export K3S_DATASTORE_ENDPOINT='%s'
+curl -sfL https://get.k3s.io | sh -s - server --token=$SECRET --node-taint CriticalAddonsOnly=true:NoExecute --tls-san %s
+`, token, dbEndpoint, privateIPlb)
+}
+
+func scriptKUBECONFIG() string {
+	return `#!/bin/bash
+cat /etc/rancher/k3s/k3s.yaml`
+}
+
+func FetchKUBECONFIG(instanceCP *civogo.Instance) (string, error) {
+	kubeconfig, err := ExecWithOutput(instanceCP.PublicIP, instanceCP.InitialPassword, scriptKUBECONFIG(), true)
+	if err != nil {
+		return "", nil
+	}
+	return kubeconfig, nil
+}
+
+func CreateControlPlane(client *civogo.Client, number int, clusterName, nodeSize string) (*civogo.Instance, error) {
+
+	network, err := GetNetwork(client, clusterName+"-ksctl")
+	if err != nil {
+		return nil, err
+	}
+
+	diskImg, err := client.GetDiskImageByName("ubuntu-focal")
+	if err != nil {
+		return nil, err
+	}
+
+	name := fmt.Sprintf("%s-ksctl-cp-%d", clusterName, number)
```

Comment:
confirm if going with randomized naming convention a better way inplace of sequence

if we are going with random make sure to remove parameters `int`